splatter.py

```diff
@@ -0,0 +1,32 @@
+
+import uuid
+from pymongo import MongoClient
+
+api_key_length = 30
+
+
+class SplatterService(object):
+
+    def __init__(self, database):
+        self.database = database
+
+    def create_api_key(self, name, key=None):
+
+        if key is not None and len(str(key)) != api_key_length:
+            raise ValueError("API key must be %s characters long")
+
+        if key is None:
+            key = uuid.uuid4()
+            key = str(key)[:30]
```

Comment:
Actually you know what let's just use the standard 38 char 128-bit uuid generated by `uuid.uuid4()` - still allow custom ones if we want but I don't think we have any real excuse for trimming a valid UUID